### DISCUSSION TITLE
How to use the github-actions bot

### DISCUSSION THREAD

**Discussion by lionel-faber** [community/community#25863] [Actions]
<p>I’ve created a couple of GitHub actions that add labels to an issue and assign it to a user. But, after the action executes the issue reflects that <code>user-a</code> has added labels <code>a</code> and <code>b</code>. This is probably because I’m using <code>user-a</code>s API token to authorise the adding of theses labels.</p>
<p>I’m looking for a way to authorise the <code>github-actions[bot]</code> to perform these operations so that users would know that they were triggered by automation.</p>
<p>Any help / links are appreciated. Thanks!</p>

**Comment by oldskool** [comment, marked as answer]
<p>The GitHub Actions bot already has the permissions you want by default. Simply call the GitHub API using the token that is generated by GitHub Actions in the <code>GITHUB_TOKEN</code> secret. For example:</p>
<pre><code>jobs:
  assign:
    runs-on: ubuntu-latest

steps:
      - name: Assign user
        env:
          PR_NUMBER: ${{ github.event.pull_request.number }}
        run: |
          curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -X POST -d '{"assignees": ["oldskool"]}' \
          https://api.github.com/repos/${GITHUB_REPOSITORY}/issues/${PR_NUMBER}/assignees
</code></pre>
<p>Will result in the github-actions bot assigning me to the PR that triggered this workflow:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c651d848d36ce5cfa9ffaf75d69ac6e945a7bb45.png" data-download-href="/uploads/short-url/sipTePCglqEEhix189lOqDTTqqV.png?dl=1" title="Screenshot 2019-10-10 at 16.05.46.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/567664/181093764-6688666d-f00c-437c-9858-98c2a0d4636d.png" alt="Screenshot 2019-10-10 at 16.05.46.png" data-base62-sha1="sipTePCglqEEhix189lOqDTTqqV" width="690" height="308" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c651d848d36ce5cfa9ffaf75d69ac6e945a7bb45_2_690x308.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/c651d848d36ce5cfa9ffaf75d69ac6e945a7bb45_2_1035x462.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/c/c651d848d36ce5cfa9ffaf75d69ac6e945a7bb45_2_1380x616.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c651d848d36ce5cfa9ffaf75d69ac6e945a7bb45_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2019-10-10 at 16.05.46.png</span><span class="informations">1508×674 68.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by lionel-faber** [comment]
<p>Thanks so much <a class="mention" href="/oldskool">@oldskool</a> </p>
<p>I had generated an API token and was using that to authenticate the Github API.<br>
I was not aware of the default API token available already!</p>
